source/speech/speech.py

```diff
@@ -164,11 +164,13 @@ def processText(locale: str, text: str, symbolLevel: characterProcessing.SymbolL
 	return text.strip()
 
 
-def cancelSpeech():
+def cancelSpeech(clearBrailleRegions=True):
 	"""Interupts the synthesizer from currently speaking"""
 	# Import only for this function to avoid circular import.
 	from .sayAll import SayAllHandler
 	SayAllHandler.stop()
+	if clearBrailleRegions:
+		_regions.clear()

```

Comment:
it seems like _regions is only cleared when speech is cancelled. This means the _regions list could grow infinitely if a user continues to use NVDA. It should be cleared when any speech sequence ends naturally as well.